ray-curator/tests/stages/deduplication/fuzzy/test_minhash.py

```diff
@@ -0,0 +1,185 @@
+"""Test suite for GPUMinHash functionality."""
+
+from collections.abc import Iterable
+from itertools import combinations
+
+import cudf
+import numpy as np
+import pandas as pd
+import pytest
+
+from ray_curator.stages.deduplication.fuzzy.minhash import GPUMinHash
+
+
+def minhash_overlap(minhash1: np.ndarray, minhash2: np.ndarray) -> float:
+    """Calculate the overlap ratio between two minhash signatures."""
+    assert len(minhash1) == len(minhash2)
+    overlap = sum(minhash1 == minhash2)
+    return overlap / len(minhash1)
+
+
+def jaccard_index(str1: str, str2: str, char_ngrams: int) -> float:
+    """Calculate the true Jaccard index between two strings."""
+    return cudf.Series([str1]).str.jaccard_index(cudf.Series([str2]), width=char_ngrams).values_host[0]
+
+
+def generate_all_pairs(item: Iterable) -> Iterable:
+    """Generate all pairs of items from an iterable."""
+    return combinations(item, 2)
+
+
+@pytest.fixture
+def sample_data():
+    """Create sample test data."""
+    return pd.DataFrame(
+        {
+            "id": [1, 2, 300, 4, -1],
+            "text": [
+                "A test string",
+                "A different test string",
+                "A different object",
+                "The quick brown fox jumps over the lazy dog",
+                "The quick black cat jumps over the lazy dog",
+            ],
+        }
+    )
+
+
+@pytest.mark.gpu
```

Comment:
Ci tests seem to be failing because of this not being recognized. Do I need to add the pytest config as well?